integration_test/validate_metadata_test.go

```diff
@@ -18,7 +20,42 @@ var agentMetricsMetadata []byte
 var thirdPartyDataDir embed.FS
 
 func TestValidateMetadataOfThirdPartyApps(t *testing.T) {
-	err := fs.WalkDir(thirdPartyDataDir, ".", func(path string, info fs.DirEntry, err error) error {
+	err := walkThirdPartyApps(func(contents []byte) error {
+		return validateMetadata(contents, &metadata.IntegrationMetadata{})
+	})
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestThirdPartyPublicUrls(t *testing.T) {
+	err := walkThirdPartyApps(func(contents []byte) error {
+		integrationMetadata := &metadata.IntegrationMetadata{}
+		err := validateMetadata(contents, integrationMetadata)
+		if err != nil {
+			return err
+		}
+		t.Run(integrationMetadata.ShortName, func(t *testing.T) {
+			t.Parallel()
+			r, err := http.Get(integrationMetadata.PublicUrl)
+			if err != nil {
+				t.Error(err)
+			}
+			if r.StatusCode == 404 {
+				t.Error(fmt.Sprintf("Invalid public url: %s", integrationMetadata.PublicUrl))
+			}
+			fmt.Println(r)
+		})
+		return nil
+	})
+
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func walkThirdPartyApps(fn func(contents []byte) error) error {
+	return fs.WalkDir(thirdPartyDataDir, ".", func(path string, info fs.DirEntry, err error) error {
 		if info.Name() != "metadata.yaml" {
```

Comment:
[Not directly related to this PR] Does `if info.Name() != "metadata.yaml" {` mean the `metadata.yaml` file is not present for this application. Do we need to error out in that case?

Reply:
The function will be called for every directory entry under the given path. This conditional just means we'd skip any file that isn't named "metadata.yaml".

Reply:
`fs.WalkDir()` goes through all directories in `third_party_apps_data` and will also be called on all files that exist within each directory. There is more than just `metadata.yaml` files present so `if info.Name() != "metadata.yaml" {` serves as a filter to only address `metadata.yaml` files. 

We could put a test case that checks `third_party_apps_data/applications/<application_name>/metadata.yaml` exists.

Reply:
Ah i see. Makes sense.

Will the test fail if no `metadata.yaml` file exists?

Reply:
The newly purposed test would fail, if a `metadata.yaml` file does not exist. 

Currently for `walkThirdPartyApps()` if there is a missing `metadata.yaml` in a particular directory, said directory would be skipped.

Reply:
SG